nestjs/src/courses/course-mentors/course-mentors.service.ts

```diff
@@ -0,0 +1,191 @@
+      `,
+        { courseId, searchText },
+      )
+      .limit(20)
+      .getMany();
+
+    const result = entities.map(entity => ({
+      id: entity.id,
+      githubId: entity.user.githubId,
+      name: PersonDto.getName(entity.user),
+    }));
+
+    return result;
+  }
+
+  private async getMentorsRecordsWithDetails(courseId: number) {
+    return await this.mentorsRepository
+      .createQueryBuilder('mentor')
+      .innerJoin('mentor.user', 'user')
+      .addSelect(UsersService.getPrimaryUserFields())
+      .addSelect(['user.contactsEpamEmail'])
+      .leftJoin('mentor.students', 's')
+      .leftJoin('mentor.stageInterviews', 'si')
+      .leftJoin('mentor.taskChecker', 'tc')
+      .leftJoin('mentor.interviewResults', 'ir')
+      .leftJoin('tc.courseTask', 't', 't.type = :type', { type: 'interview' })
+      .addSelect(['s.id', 's.isExpelled', 's.isFailed'])
+      .addSelect(['si.id', 'si.isCompleted'])
+      .addSelect(['tc.id', 'tc.courseTaskId'])
+      .addSelect(['t.id', 't.type'])
+      .addSelect(['ir.id'])
+      .where(`"mentor"."courseId" = :courseId`, { courseId })
+      .orderBy('mentor.createdDate')
+      .getMany();
+  }
+
+  private buildTaskResultSubQuery(courseTasksIds: number[]) {
+    const subQuery = this.taskResultRepository.createQueryBuilder('t').select('t.lastCheckerId', 'lastCheckerId');
+
+    if (courseTasksIds.length) {
+      subQuery.where('t.courseTaskId IN (:...ids)', { ids: courseTasksIds });
+    } else {
+      // to prevent empty array
+      subQuery.where('1 = 0');
+    }
+
+    return subQuery;
+  }
+
+  private async getLastCheckedDates(courseId: number, courseTasksIds: number[]): Promise<Record<string, Date | null>> {
+    const subQuery = this.buildTaskResultSubQuery(courseTasksIds)
+      .addSelect('MAX(t."updatedDate")', 'updatedDate')
+      .groupBy('t.lastCheckerId');
+
+    const query = this.dataSource
+      .createQueryBuilder()
+      .select('m.id', 'id')
+      .addSelect('tr."updatedDate"', 'value')
+      .from(Mentor, 'm')
+      .leftJoin('user', 'u', 'u."id" = m."userId"')
+      .leftJoin('(' + subQuery.getQuery() + ')', 'tr', 'tr."lastCheckerId" = u.id')
+      .where('m."courseId" = :courseId', { courseId })
+      .setParameters(subQuery.getParameters());
+
+    const result: { id: number; value: Date | null }[] = await query.getRawMany();
+
+    return Object.fromEntries(result.map(({ id, value }) => [id, value]));
+  }
+
+  private async getCheckedTasksCount(courseId: number, courseTasksIds: number[]): Promise<Record<string, number>> {
+    const subQuery = this.buildTaskResultSubQuery(courseTasksIds).addSelect('t."updatedDate"', 'updatedDate');
+
+    const query = this.dataSource
+      .createQueryBuilder()
+      .select('m.id', 'id')
+      .addSelect('COUNT(tr."lastCheckerId")', 'value')
+      .from(Mentor, 'm')
+      .leftJoin('user', 'u', 'u."id" = m."userId"')
+      .leftJoin('(' + subQuery.getQuery() + ')', 'tr', 'tr."lastCheckerId" = u.id')
```

Comment:
could you please switch to template string here and in same places in this file?

Reply:
Sure. Thank you

Reply:
Fixed